File: 0100.same_tree/solution.py
from utils import TreeNode


class Solution:
    def is_same_tree(self, p: TreeNode, q: TreeNode) -> bool:
        if p is None and q is None:
            return True
        if not p or not q:
            return False
        return p.val == q.val and self.is_same_tree(p.left, q.left) and self.is_same_tree(p.right, q.right)

File: 0100.same_tree/solution.go
package leetcode0100

import "github.com/Ackerr/Algorithms/utils"

func isSameTree(m *utils.TreeNode, n *utils.TreeNode) bool {
	if m == nil && n == nil {
		return true
	} else if m == nil || n == nil {
		return false
	}
	return m.Val == n.Val && isSameTree(m.Left, n.Left) && isSameTree(m.Right, n.Right)
}
